### ISSUE TITLE
Replicate the Adverserial Autoencoder Results as stated in the Paper

### ISSUE THREAD

**Issue by yashkotadia** [uvavision/Balanced-Datasets-Are-Not-Enough#3]
Hi,

I was hoping if you could delineate the exact steps required to replicate the results in the paper for adv@image for COCO.

I have been able to achieve the following results with the given training settings:

--adv_lambda 5.0 --batch_size 16 --beta 0.1 --batch_balanced --num_epochs 35 --learning_rate 0.00001 --autoencoder_finetune --adv_on --resume

I have also finetuned the gender classifier from scratch with the weights of Resnet18 in the convolutional layers.

Train epoch 35: meanAP: 46.42, adv acc: 70.02
Test epoch(f): 35, meanAP: 42.99, adv acc: 66.67
Test epoch   : 35, meanAP: 42.05, adv acc: 67.86

The mAP for adv@image stated in the paper is 56.14
Am I doing something wrong? Or should I be training for more epochs(although it doesn't seem like that would help because the model has been stuck at these accuracy levels since 5 epochs)

**Comment by tianlu-wang** [comment]
Could you please try first pretrain the multi-label model on normal images? Please refer to commands.ipynb. I think what I did was first tune the Unet and then tune the multi-label model (resnet50). You probably need to stop early to maintain the performance, if you keep training the multi-label performance will drop.

**Comment by yashkotadia** [comment]
Can you explain more clearly at each step which all components to train and which all weights are to be frozen?

**Comment by yashkotadia** [comment]
You need to have a pretrained resnet50 for multi-label and a pretrained resnet18 for gender classification. 
1. Freeze everything except the Unet, the performance will go down from 53.75(F1 score) to some point.
2. Freeze the Unet and tune other parts, the performance will go up.

for example: 
1. CUDA_VISIBLE_DEVICES=3 python ae_adv_train.py --save_dir 6 --log_dir 6 --layer generated_image --adv_lambda 5.0 --batch_size 16 --beta 0.1 --batch_balanced --num_epochs 80 --learning_rate 0.00001 --autoencoder_finetune --adv_on --resume
2. CUDA_VISIBLE_DEVICES=3 python ae_adv_train.py --save_dir 6_75 --log_dir 6_75 --layer generated_image --adv_lambda 5.0 --batch_size 16 --beta 0.1 --batch_balanced --num_epochs 160 --learning_rate 0.00001 --resume --finetune --checkpoint checkpoint_75.pth.tar

**Comment by tianlu-wang** [comment]
I am closing this issue as I have explained in emails with you. Feel free to reopen if you still have questions.
